src/world.cpp

```diff
@@ -120,3 +128,59 @@ World::get_mesh_greedy() const {
     }
     return out;
 }
+
+void
+World::update_all_chunks_mesh() {
+    for (size_t i = 0; i < chunks_mesh.size(); i++)
+        update_single_mesh(i);
+}
+
+void
+World::update_single_mesh(uint16_t chunk_pos) {
+    const auto& chunks = terrain_main.get_chunks();
+    entity::Mesh chunk_mesh = entity::generate_mesh(chunks[chunk_pos]);
+
+    chunk_mesh.change_color_indexing(
+        materials, terrain::TerrainColorMapping::get_colors_inverse_map()
+    );
+
+    chunks_mesh[chunk_pos]->update(chunk_mesh);
+}
+
+void
+World::update_single_mesh(TerrainDim3 tile_sop) {
+    if (terrain_main.in_range(tile_sop.x + 1, tile_sop.y, tile_sop.z)) {
+        uint16_t chunk_pos =
+            terrain_main.get_chunk_from_tile(tile_sop.x + 1, tile_sop.y, tile_sop.z);
+        update_single_mesh(chunk_pos);
+    }
+}
```

Comment:
define these in a sensical order

```suggestion
void
World::update_single_mesh(uint16_t chunk_pos) {
    const auto& chunks = terrain_main.get_chunks();
    entity::Mesh chunk_mesh = entity::generate_mesh(chunks[chunk_pos]);

    chunk_mesh.change_color_indexing(
        materials, terrain::TerrainColorMapping::get_colors_inverse_map()
    );

    chunks_mesh[chunk_pos]->update(chunk_mesh);
}

void
World::update_single_mesh(TerrainDim3 tile_sop) {
    if (terrain_main.in_range(tile_sop.x + 1, tile_sop.y, tile_sop.z)) {
        uint16_t chunk_pos =
            terrain_main.get_chunk_from_tile(tile_sop.x + 1, tile_sop.y, tile_sop.z);
        update_single_mesh(chunk_pos);
    }
}

void
World::update_all_chunks_mesh() {
    for (size_t i = 0; i < chunks_mesh.size(); i++)
        update_single_mesh(i);
}
```